src/backend/tests/reimbursement-requests.test.ts

```
@@ -259,4 +259,55 @@ describe('Reimbursement Requests', () => {
       expect(res).toStrictEqual([]);
     });
   });
+
+  describe('Delivered Tests', () => {
+    test('Mark as delivered fails for non submitter', async () => {
+      jest.spyOn(prisma.reimbursement_Request, 'findUnique').mockResolvedValue(GiveMeMyMoney);
+
+      await expect(
+        ReimbursementRequestService.markReimbursementRequestAsDelivered(wonderwoman, GiveMeMyMoney.reimbursementRequestId)
+      ).rejects.toThrow(new AccessDeniedException('Only the creator of the reimbursement request can mark as delivered'));
+
+      expect(prisma.reimbursement_Request.findUnique).toBeCalledTimes(1);
+    });
+
+    test('Mark as delivered fails for undefined ID', async () => {
+      jest.spyOn(prisma.reimbursement_Request, 'findUnique').mockResolvedValue(GiveMeMyMoney);
+
+      await expect(ReimbursementRequestService.markReimbursementRequestAsDelivered(batman, '1234')).rejects.toThrow(
```

it doesn't actually matter what id you pass in here for the requestId, if you look at the service function, it's checking for the existence of a reimbursementRequest gotten from findUnique. For that to be null and throw the error you want, you need to mock it to be null above instead of mocking it as GiveMeMyMoney.

---

you can pass in GiveMeMyMoney instead of 1234 if you want the error messages to match up